src/main/java/com/example/webapp/Service/DepartmentService.java

```diff
@@ -0,0 +1,71 @@
+package com.example.webapp.Service;
+
+import com.example.webapp.Model.Department;
+import com.example.webapp.Repository.DepartmentRepository;
+import com.example.webapp.dto.DepartmentDTO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class DepartmentService {
+    @Autowired
+    DepartmentRepository departmentRepository;
+
+    public List<DepartmentDTO> getAll() {
+        List<DepartmentDTO> departmentList = new ArrayList<>();
+        departmentRepository.findAll().forEach(department -> departmentList.add(convertENTtoDTO(department)));
+        if (departmentList.isEmpty()) {
+            return Collections.emptyList();
+        }
+        return departmentList;
+    }
+
+    public DepartmentDTO getById(long id) {
+        Optional<Department> dep = departmentRepository.findById(id);
+        if(dep.isPresent()){
+            return convertENTtoDTO(dep.get());
+        }
+        return null;
+    }
+
+    public DepartmentDTO add(DepartmentDTO department) {
+        departmentRepository.save(convertDTOtoENT(department));
+        return department;
+    }
+
+    public DepartmentDTO updateDepartment(long id, DepartmentDTO department) {
+        Optional<Department> departmentData = departmentRepository.findById(id);
+        if (departmentData.isPresent()) {
+            Department newDep = departmentData.get();
+            newDep.setName(department.getName());
+            newDep.setLocation(department.getLocation());
+            departmentRepository.save(newDep);
+            return convertENTtoDTO(newDep);
+        } else {
+            return null;
+        }
+    }
+
+    public Department convertDTOtoENT(DepartmentDTO departmentDTO) {
+        return Department.builder()
+                .department_id(departmentDTO.getId())
+                .name(departmentDTO.getName())
+                .location(departmentDTO.getLocation())
+                .employeeSet(departmentDTO.getEmployeeSet())
+                .build();
+    }
+
+    public DepartmentDTO convertENTtoDTO(Department department) {
```

Comment:
make static and move to DepartmentDTO pojo

Reply:
Done